eng/scripts/Language-Settings.ps1

```diff
@@ -5,6 +5,58 @@ $packagePattern = "*.zip"
 $MetadataUri = "https://raw.githubusercontent.com/Azure/azure-sdk/master/_data/releases/latest/python-packages.csv"
 $BlobStorageUrl = "https://azuresdkdocs.blob.core.windows.net/%24web?restype=container&comp=list&prefix=python%2F&delimiter=%2F"
 
+function Get-AllPackageInfoFromRepo ($serviceDirectory)
+{
+  $allPackageProps = @()
+  $searchPath = "sdk/*/*/setup.py"
+  if ($serviceDirectory)
+  {
+    $searchPath = "sdk/${serviceDirectory}/*/setup.py"
+  }
+
+  $allPkgPropLines = $null
+  try
+  {
+    Push-Location $RepoRoot
+    pip install packaging==20.4 -q -I
+    $allPkgPropLines = python "eng\scripts\get_package_properties.py" -s $searchPath
+  }
+  catch
+  {
+    # This is soft error and failure is expected for python metapackages
+    LogError "Failed to get all package properties"
+  }
+  finally
+  {
+    Pop-Location
+  }
+
+  foreach ($line in $allPkgPropLines)
+  {
+    $pkgInfo = ($line -Split ",").Trim("()' ")
+    $packageName = $pkgInfo[0]
+    $packageVersion = $pkgInfo[1]
+    $isNewSdk = $pkgInfo[2]
+    $setupPyDir = $pkgInfo[3]
+    $pkgDirectoryPath = Resolve-Path (Join-Path -Path $RepoRoot $setupPyDir)
+    $serviceDirectoryName = Split-Path (Split-Path -Path $pkgDirectoryPath -Parent) -Leaf
```

Comment:
@scbedd Can we guarantee the setup.py is always at this relative path? It might be better to have the find the "sdk".